v2/delay/delay_test.go

```diff
@@ -228,6 +235,39 @@ func TestRunningFunction(t *testing.T) {
 	}
 }
 
+func TestRunningRegister(t *testing.T) {
```

Comment:
This test suite currently has 10 `delay.Func` tests, and now we're adding one `delay.Register` test.  Given that `Register` is the new well-lit path, that seems a bit backwards.

Can we make most of these tests run 2 subtests, one for `Func` and one for `Register`?  Maybe you can make a test helper that handles that and can be reused in most of the tests.